src/crypto.ts

```diff
@@ -42,31 +48,52 @@ export interface AsyncVerifier<A extends number> {
 }
 
 /**
- * Just like Verifier` but definitely sync.
+ * Just like {@link Verifier}, but definitely sync.
  */
 export interface SyncVerifier<A extends number> {
   verify<T>(payload: ByteView<T>, signature: Signature<T, A>): boolean
 }
 
+/**
+ * Just like {@link Signer}, but definitely sync.
+ */
 export interface SyncSigner<A extends number = number> {
   sign<T>(payload: ByteView<T>): Signature<T, A>
 }
 
+/**
+ * Just like {@link Signer}, but definitely async.
+ */
 export interface AsyncSigner<A extends number = number> {
   sign<T>(payload: ByteView<T>): PromiseLike<Signature<T, A>>
 }
 
 /**
- * Represents byte encoded representation of the `Data`. It uses type parameter
- * to capture the structure of the data it encodes.
+ * A byte-encoded representation of some type of `Data`. 
+ * 
+ * A `ByteView` is essentially a `Uint8Array` that's been "tagged" with
+ * a `Data` type parameter indicating the type of encoded data.
+ * 
+ * For example, a `ByteView<DID>` is a `Uint8Array` containing a binary
+ * representation of a {@link DID}.
  */
 export interface ByteView<Data> extends Uint8Array, Phantom<Data> {}
 
+/**
+ * Something you can `await` and get a `T` out of. Either a `T` already, or a Promise for a `T`.
+ */
 export type Await<T> = T | PromiseLike<T>
 
 /**
- * This is an utility type to retain unused type parameter `T`. It can be used
- * as nominal type e.g. to capture semantics not represented in actual type strucutre.
+ * A utility type to retain an unused type parameter `T`.
+ * 
+ * Capturing unused type parameters allows us to define "nominal types," which
+ * TypeScript does not natively support. Nominal types in turn allow us to capture
+ * semantics not represented in the actual type structure, without requring us to define
+ * new classes or pay additional runtime costs.
+ * 
+ * For a concrete example, see {@link ByteView}, which extends the `Uint8Array` type to capture
+ * type information about the structure of the data encoded into the array.
```

Comment:
Do you think it might be worth linking to https://doc.rust-lang.org/rust-by-example/generics/phantom.html ?